crates/holochain/src/conductor/tests/cell_cloning.rs

```diff
@@ -196,6 +171,127 @@ async fn app_info_includes_cloned_cells() {
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn clone_cell_deletion() {
+    let (dna, _, _) = SweetDnaFile::unique_from_test_wasms(vec![TestWasm::Create])
+        .await
+        .unwrap();
+    let role_id: AppRoleId = "dna_1".to_string();
+    let mut conductor = SweetConductor::from_standard_config().await;
+    let alice = SweetAgents::one(conductor.keystore()).await;
+    let app_id = "app";
+    conductor
+        .setup_app_for_agent(app_id, alice.clone(), [&(role_id.clone(), dna)])
+        .await
+        .unwrap();
+    let installed_clone_cell = conductor
+        .clone()
+        .create_clone_cell(CreateCloneCellPayload {
+            app_id: app_id.to_string(),
+            role_id: role_id.clone(),
+            phenotype: DnaPhenotypeOpt::none().with_network_seed("seed_1".to_string()),
+            membrane_proof: None,
+            name: None,
+        })
+        .await
+        .unwrap();
+
+    // archive clone cell
+    conductor
+        .inner_handle()
+        .archive_clone_cell(ArchiveCloneCellPayload {
+            app_id: app_id.to_string(),
+            clone_cell_id: CloneCellId::CloneId(
+                CloneId::try_from(installed_clone_cell.clone().into_role_id()).unwrap(),
+            ),
+        })
+        .await
+        .unwrap();
+
+    // assert that the cell doesn't appear any longer in app info's cell data
+    let app_info = conductor
+        .get_app_info(&app_id.to_string())
+        .await
+        .unwrap()
+        .unwrap();
+    assert_eq!(app_info.cell_data.contains(&installed_clone_cell), false);
+
+    // calling the cell after archiving fails
+    let zome = SweetZome::new(
+        installed_clone_cell.as_id().clone(),
+        TestWasm::Create.coordinator_zome_name(),
+    );
+    let zome_call_response: Result<ActionHash, _> = conductor
+        .call_fallible(&zome, "call_create_entry", ())
+        .await;
+    assert!(zome_call_response.is_err());
+
+    // restore the archived clone cell
+    let restored_cell = conductor
+        .inner_handle()
+        .restore_archived_clone_cell(ArchiveCloneCellPayload {
+            app_id: app_id.into(),
+            clone_cell_id: CloneCellId::CloneId(
+                CloneId::try_from(installed_clone_cell.clone().into_role_id()).unwrap(),
+            ),
+        })
+        .await
+        .unwrap();
+
+    // assert the restored clone cell is the previously created clone cell
+    assert_eq!(restored_cell, installed_clone_cell);
+
+    // assert that the cell appears in app info's cell data again
+    let app_info = conductor
+        .get_app_info(&app_id.to_string())
+        .await
+        .unwrap()
+        .unwrap();
+    assert_eq!(app_info.cell_data.contains(&installed_clone_cell), true);
+
+    // calling the cell after restoring succeeds
+    let zome = SweetZome::new(
+        installed_clone_cell.as_id().clone(),
+        TestWasm::Create.coordinator_zome_name(),
+    );
```

Comment:
Would be nice to define this zome only once to make it even clearer that the two calls are exactly the same. Not necessary though.